website_geonames/static/src/js/website_geonames_field.js

```diff
@@ -0,0 +1,47 @@
+odoo.define('website_geonames', function () {
+    $(document).ready(function () {
```

Comment:
In Odoo, the right way to do this is to register a publicWidget that with a jquery selector to target an element
The public widget will be instantiated on any page the selector is found

```js
const publicWidget = require('web.public.widget');

publicWidget.registry.geonamesAutocomplete = publicWidget.Widget.extend({
    selector: "" // <--- target here the form that contains the city, zip, etc.. 

    /**
     * @override
     */
    start: function () {
        // do somethign here to find the form children you expect
        this.$zip = this.$el.find(".zip_id")
        // same for city, state, etc..
        // also take this opportunity to initialize the jquery plugin
        // basically this method is called once when the selector element is found, and it's meant to initliaze the widget
    },
});
```


in the odoo code base, search for `publicWidget.registry` and you'll find plenty of examples for inspiration